src/signer/op-signer/launcher.star

```diff
@@ -0,0 +1,272 @@
+        config=config_file_name,
+    )
+
+
+# Creates a map of private keys keyed by client hostnames
+def _create_private_keys(plan, params, clients, script_artifacts, registry):
+    return {
+        client.hostname: _create_private_key(
+            plan=plan,
+            params=params,
+            client=client,
+            script_artifacts=script_artifacts,
+            registry=registry,
+        )
+        for client in clients
+    }
+
+
+# Converts a HEX private key to a PEM format
+def _create_private_key(plan, params, client, script_artifacts, registry):
+    key_file_name = "{}.pem".format(client.hostname)
+    key_file_path = "/tmp/{}".format(key_file_name)
+
+    convert_private_key = plan.run_sh(
+        name="{}-convert-pk-{}".format(params.service_name, client.hostname),
+        description="Convert private key for {} to PEM format".format(client.hostname),
+        run="/scripts/convert-private-key.sh {} > {}".format(
+            client.private_key, key_file_path
+        ),
+        image=registry.get(_registry.OPENSSL),
+        files={"/scripts": script_artifacts},
+        store=[
+            StoreSpec(
+                src=key_file_path,
+                name="{}--pem--{}".format(params.service_name, client.hostname),
+            ),
+        ],
+    )
+
+    return struct(
+        # We'll return the private key artifact itself
+        artifact=convert_private_key.files_artifacts[0],
+        # As well as the relative file path
+        key=key_file_name,
+    )
+
+
+# Helper function that uploads all the shell scripts we'll need for the signer launch
+def _create_script_artifacts(plan, params):
+    # First we upload the local shell script that generates the credentials
+    return plan.upload_files(
+        src="./scripts",
+        name="{}--scripts".format(params.service_name),
+    )
+
+
+# Helper function to assert client struct validity
+def _assert_client(client):
+    typeof_client = type(client)
+    if typeof_client != "struct":
+        fail("op-signer: client must be a struct, got {}".format(typeof_client))
+
+    if not hasattr(client, "hostname"):
+        fail(
+            "op-signer: client struct must have a 'hostname' field, got {}".format(
+                client
+            )
+        )
+
+    typeof_hostname = type(client.hostname)
+    if typeof_hostname != "string":
+        fail(
+            "op-signer: client struct 'hostname' field must be a string, got {}".format(
+                typeof_hostname
+            )
+        )
+
+    if not hasattr(client, "private_key"):
+        fail(
+            "op-signer: client struct must have a 'private_key' field, got {}".format(
+                client
+            )
+        )
+
+    typeof_private_key = type(client.private_key)
+    if typeof_private_key != "string":
+        fail(
+            "op-signer: client struct 'private_key' field must be a string, got {}".format(
+                typeof_private_key
+            )
+        )
```

Comment:
Should we add a check to ensure the private key is correctly formatted (i.e. not empty string, expected number of characters) so there are not any confusing downstream errors?